builder/azure/common/template/template_builder_test.go

```diff
@@ -181,3 +181,42 @@ func TestBuildWindows02(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// Shared Image Gallery Build
+func TestSharedImageGallery00(t *testing.T) {
+	testSubject, err := NewTemplateBuilder(BasicTemplate)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = testSubject.BuildLinux("--test-ssh-authorized-key--")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	imageID := "/subscriptions/ignore/resourceGroups/ignore/providers/Microsoft.Compute/galleries/ignore/images/ignore"
+	err = testSubject.SetSharedGalleryImage("westcentralus", imageID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if (*testSubject.template.Variables)["apiVersion"] != "2018-04-01" {
+		t.Fatal("ARM template for Shared Image Gallery must use apiVersion 2018-04-01")
+	}
+
+	foundImageID := false
```

Comment:
I was expecting more a test like TestBuildLinux02, which uses the approval library.  You create a JSON document using the NewTemplateBuilder, and serialize it to disk. The approval library verifies that the output matches the golden copy that is checked into source control.